src/Sulu/Bundle/ContentBundle/Controller/VersionController.php

```diff
@@ -0,0 +1,183 @@
+use Sulu\Component\Rest\ListBuilder\ListRepresentation;
+use Sulu\Component\Rest\RequestParametersTrait;
+use Sulu\Component\Security\Authorization\AccessControl\SecuredObjectControllerInterface;
+use Sulu\Component\Security\SecuredControllerInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Handles the versions of pages.
+ */
+class VersionController extends FOSRestController implements
+    ClassResourceInterface,
+    SecuredControllerInterface,
+    SecuredObjectControllerInterface
+{
+    use RequestParametersTrait;
+
+    /**
+     * Returns the versions for the node with the given UUID.
+     *
+     * @param Request $request
+     * @param string $uuid
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function cgetAction(Request $request, $uuid)
+    {
+        $locale = $this->getRequestParameter($request, 'language', true);
+
+        $document = $this->get('sulu_document_manager.document_manager')->find($uuid, $request->query->get('language'));
+        $versions = array_reverse(array_filter($document->getVersions(), function($version) use ($locale) {
+            /** @var Version $version */
+            return $version->getLocale() === $locale;
+        }));
+        $total = count($versions);
+
+        $listRestHelper = $this->get('sulu_core.list_rest_helper');
+        $limit = $listRestHelper->getLimit();
+
+        $versions = array_slice($versions, $listRestHelper->getOffset(), $limit);
+
+        $userIds = array_unique(array_map(function($version) {
+            /** @var Version $version */
+            return $version->getAuthor();
+        }, $versions));
+
+        $users = $this->get('sulu_security.user_repository')->findUsersById($userIds);
+        $fullNamesByIds = [];
+        foreach ($users as $user) {
+            $fullNamesByIds[$user->getId()] = $user->getContact()->getFullName();
+        }
+
+        $versionData = [];
+        foreach ($versions as $version) {
+            $versionData[] = [
+                'id' => str_replace('.', '_', $version->getId()),
+                'locale' => $version->getLocale(),
+                'author' => array_key_exists($version->getAuthor(), $fullNamesByIds)
+                    ? $fullNamesByIds[$version->getAuthor()] : '',
+                'authored' => $version->getAuthored(),
+            ];
+        }
+
+        $versionCollection = new ListRepresentation(
+            $versionData,
+            'versions',
+            'get_node_versions',
+            ['uuid' => $uuid],
+            $listRestHelper->getPage(),
+            $limit,
+            $total
+        );
+
+        return $this->handleView($this->view($versionCollection));
+    }
+
+    /**
+     * @param Request $request
+     * @param string $uuid
+     * @param int $version
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function postAction(Request $request, $uuid, $version)
```

Comment:
The cool thing here is that we don't need the `@Post` annotation as we have in the `NodeController` the `$version` parameter will also be added to the route from the method signature.

The bad thing is that the security is not working correctly then, since it is recognizing the `POST` as create, which means that every who is allowed to add a page can also restore it...

That's probably part of a bigger issue, we've got several ways to handle this.

1. Find a different way to distinguish between `add` and `create` in security (would have to think about that, maybe if the last part excluding the query string is number or uuid?)
2. Handle security without the fancy helpers Sulu gives us.
3. Rename that action to `postTrigger` action as we are used to it, and add a `@Post` annotation. The drawback is that we have to include the parent route in the annotation: `/nodes/{uuid}/versions/{version}`